packages/vscode-lit-plugin/package.json

```diff
@@ -25,31 +25,120 @@
 	"main": "./out/extension",
 	"scripts": {
 		"clean": "rimraf out",
-		"vscode:prepublish": "npm run compile",
-		"compile": "npm run clean & tsc -p ./",
 		"watch": "tsc -watch -p ./",
 		"publish": "vsce publish",
-		"package": "vsce package",
+		"package": "wireit",
 		"install:safe": "npm install --ignore-scripts --no-package-lock && npm i --no-package-lock",
 		"readme": "readme generate -i readme.blueprint.md -c readme.config.json",
-		"test": "npm run compile && node ./out/test/scripts/test-runner.js",
-		"test-packaged": "vsce package -o ./out/packaged.vsix && rm -rf ../../../packaged-extension/ && mkdir ../../../packaged-extension/ && unzip -qq ./out/packaged.vsix -d ../../../packaged-extension/ && node ./out/test/scripts/test-runner.js ../../../packaged-extension/extension/"
+		"test": "wireit",
+		"build": "wireit",
+		"test:normal": "wireit",
+		"test:packaged": "wireit",
+		"create-isolated-extension-dir": "wireit",
+		"test:packaged-sequentialized": "wireit"
+	},
+	"wireit": {
+		"build": {
+			"dependencies": [
+				"../ts-lit-plugin:build",
+				"../lit-analyzer:build",
+				"../../:copylink"
+			],
+			"command": "tsc --build --pretty",
+			"files": [
+				"src/**/*",
+				"tsconfig.json"
+			],
+			"output": [
+				"out/**/*.js",
+				"out/**/*.d.ts",
+				"out/**/*.d.ts.map",
+				".tsbuildinfo"
+			],
+			"clean": "if-file-deleted"
+		},
+		"test": {
+			"dependencies": [
+				"test:normal",
+				"test:packaged-sequentialized"
+			]
+		},
+		"test:normal": {
+			"dependencies": [
+				"build"
+			],
+			"files": [
+				"package.json",
+				"schemas/**/*"
+			],
+			"output": [],
+			"command": "node ./out/test/scripts/test-runner.js"
+		},
+		"test:packaged": {
+			"dependencies": [
+				"build",
+				"create-isolated-extension-dir"
+			],
+			"files": [],
+			"output": [],
+			"command": "node ./out/test/scripts/test-runner.js ../../../packaged-extension/extension/"
+		},
+		"test:packaged-sequentialized": {
+			"#comment": "like test:packaged, only it ensures that test:normal runs first",
+			"dependencies": [
+				"test:normal",
+				"build",
+				"create-isolated-extension-dir"
+			],
+			"files": [],
+			"output": [],
+			"command": "node ./out/test/scripts/test-runner.js ../../../packaged-extension/extension/"
+		},
+		"package": {
+			"dependencies": [
+				"build"
+			],
+			"files": [
+				"package.json",
+				"schemas/**/*"
+			],
+			"output": [
+				"./out/packaged.vsix"
+			],
+			"command": "vsce package -o ./out/packaged.vsix"
+		},
+		"create-isolated-extension-dir": {
+			"dependencies": [
+				"package"
+			],
+			"files": [
+				"./out/packaged.vsix"
+			],
+			"output": [
+				"../../packaged-extension/extension/**/*"
```

Comment:
Currently it's forbidden to have output outside of the package directory. It will crash when you try to do a clean build (just as an intentional check, because it seemed potentially indicative of an error) or cache (because we'll need a different directory layout to support this).

If you don't clean or cache here, it's OK though.